src/query/functions/src/aggregates/aggregate_sum.rs

```diff
@@ -271,6 +272,65 @@ where
     }
 }
 
+#[derive(BorshSerialize, BorshDeserialize, Default)]
+pub struct IntervalSumState {
+    pub value: months_days_micros,
+}
+
+impl UnaryState<IntervalType, IntervalType> for IntervalSumState {
+    fn add(
+        &mut self,
+        other: months_days_micros,
+        _function_data: Option<&dyn FunctionData>,
+    ) -> Result<()> {
+        let sum = self.value.total_micros() + other.total_micros();
+        self.value = months_days_micros(sum as i128);
+        Ok(())
+    }
+
+    fn add_batch(
+        &mut self,
+        other: Buffer<months_days_micros>,
+        validity: Option<&Bitmap>,
+        _function_data: Option<&dyn FunctionData>,
+    ) -> Result<()> {
+        let mut sum = self.value.total_micros();
```

Comment:
maybe we could implement `Add` and `AddAsign` for `months_days_micros `

Reply:
Yes.